src/analyses/ai_storage.h

```diff
@@ -211,4 +211,82 @@ class location_sensitive_storaget : public trace_map_storaget
   }
 };
 
+// The most precise form of storage
+class history_sensitive_storaget : public trace_map_storaget
+{
+protected:
+  typedef std::map<trace_ptrt, state_ptrt, ai_history_baset::compare_historyt>
```

Comment:
Do you need the ordered-ness of `map`?

Reply:
Not directly but we already have a compare function and no hash function so I figured it was better.  Also I would prefer to avoid the reallocation that comes with growing hash maps.